apm-agent-plugins/apm-es-restclient-plugin/apm-es-restclient-plugin-5_6/pom.xml

```diff
@@ -28,14 +35,13 @@
             <groupId>org.elasticsearch.client</groupId>
             <artifactId>elasticsearch-rest-high-level-client</artifactId>
             <version>${version.elasticsearch}</version>
-            <scope>provided</scope>
-        </dependency>
-
-        <!-- Common code where helper is located -->
-        <dependency>
-            <groupId>${project.groupId}</groupId>
-            <artifactId>apm-es-restclient-plugin-common</artifactId>
-            <version>${project.version}</version>
```

Comment:
just wondering, how did you found that there was some log4j files relocated in the packaged agent ? I feel that having a test that enforces allowed resource patterns in the packaged agent jar could help here.

Reply:
This is not about conflicts due to relocation. ES comes with a transitive dependency to log4j but with a different version. This lead to a `NoClassDefFoundError`.

Not sure if that's what you've asked but we test that we shade everything in `co.elastic.apm.servlet.AgentFileIT#testEverythingIsShaded`.